TriforkSwiftExtensions/TriforkSwiftExtensions/Array/ArrayExtension.swift

```diff
@@ -42,4 +87,34 @@ public extension Array {
         a.append(newElement)
         return a
     }
+
+    /// Returns the first element of the `Array`.
+    var head: Element? { first }
+    
+    /// Removes the first element and retuns a new `Array` containing the rest of the elements.
+    var tail: [Element] { Array(dropFirst()) }
+    
+    /// Returns `nil` if the `Array` is empty, otherwise returning self.
+    var nilIfEmpty: [Element]? { isEmpty ? nil : self }
+    
+    /// Splits the `Array` into two dependent on the "shouldAppendToFirst-Argument", where the first `Array` contains the elements that match the bool and the second `Array` contains the rest.
+    func split(by shouldAppendToFirst: (Element) -> Bool) -> ([Element], [Element]) {
+        var first: [Element] = []
+        var second: [Element] = []
+        
+        for element in self {
+            if shouldAppendToFirst(element) {
+                first.append(element)
+            } else {
+                second.append(element)
+            }
+        }
+        return (first, second)
+    }
+    
+    /// Return the indexed element if within boundaries, else `nil`
+    func get(index: Int) -> Element? {
```

Comment:
given the weird synergy this should potentially be called "getOrNil" to highlight the intention. that way the function is close to self documenting.